components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -88,6 +88,12 @@ func (s *Principal) GetStackTemplate() *cloudformation.Template {
 	return template
 }
 
+// GetStackPolicy implements cloudformation.Stack to return a serialised form of the stack policy, or nil if one is
+// not needed
+func (p *Principal) GetStackPolicy() string {
```

Comment:
So what we normally do for these is have a separate type that implements GetStackPolicy, then try to cast to it - if it works then GetStackPolicy was implemented and we should use it - if it doesn't then GetStackPolicy was not implemented and we'd fall back to a sane default e.g. of `""`

Reply:
Roger. I will update.

Reply:
Updated.